### ISSUE TITLE
Spike: verify Atomix membership under different cluster configurations

### ISSUE THREAD

**Issue by MiguelPires** [zeebe-io/zeebe#2006]
This issue should check that Atomix's cluster membership service works with various cluster configurations. It should also test the effect of timing issues (e.g., one node start much later than others) and network partitions on the Cluster and how Atomix deals with them.

**Comment by MiguelPires** [comment]
The most interesting scenario I tested was when broker nodes are connected in a chain (i.e., node X has X-1 as an initial contact point and node 0 has no contact point) with one gateway connected to node 0 and one "late starter", meaning the last broker in the chain starts its Atomix instance much later than the others. Regarding the gateway, I saw two requisites: 

- the gateway's Atomix instance must actually be an Atomix instance and not AtomixCluster
- the gateway must start after its initial contact point

The error we get when the gateway is the first to start depends on how its configured. If it was configured with no management partitions or data partitions, it will fail with a message "Failed to locate system partition group" or "Failed to locate primitive partition group(s)", respectively. But what's more interesting is that even if it was configured with a management and data partition, the gateway will not be connected to the cluster and vice-versa, presumably because it couldn't find its initial contact point when it started. If we start the gateway after its initial contact point, with no management and data partition, each node will find all of the others (including the gateway). Similar experiments, where one broker joins before its initial contact point, have similar results. It seems like the rule we must enforce is that before any node starts and tries to join the cluster, its contact point must have already started.
Other than that, having a broker start with a large delay doesn't cause any issues if it starts after its contact point. Starting the gateway after the cluster has fully joined and stabilized isn't a problem since it will still discover all nodes and, if it has a ClusterMembershipListener, it will receive MEMBER_ADDED for each node.
